bridges/GoogleGroupsBridge.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+class GoogleGroupsBridge extends XPathAbstract {
+	const NAME = 'Google Groups Bridge';
+	const DESCRIPTION = 'Returns the latest posts on a Google Group';
+	const URI = 'https://groups.google.com';
+	const MAINTAINER = 'Yaman Qalieh';
+	const PARAMETERS = array( array(
+		'group' => array(
+			'name' => 'Group id',
+			'title' => 'The string that follows /g/ in the URL',
+			'exampleValue' => 'announce',
+			'required' => true
+		),
+		'account' => array(
+			'name' => 'Account id',
+			'title' => 'Some Google groups have an additional id following /a/ in the URL',
+			'exampleValue' => 'mozilla.org',
+			'required' => false
+		)
+	));
+	const CACHE_TIMEOUT = 3600;
+
+	const TEST_DETECT_PARAMETERS = array(
+		'https://groups.google.com/a/mozilla.org/g/announce' => array(
+			'account' => 'mozilla.org', 'group' => 'announce'
+		),
+		'https://groups.google.com/g/ansible-project' => array(
+			'account' => null, 'group' => 'ansible-project'
+		),
+	);
+
+	const XPATH_EXPRESSION_ITEM = '//div[@class="yhgbKd"]';
+	const XPATH_EXPRESSION_ITEM_TITLE = './/span[@class="o1DPKc"]';
+	const XPATH_EXPRESSION_ITEM_CONTENT = './/span[@class="WzoK"]';
+	const XPATH_EXPRESSION_ITEM_URI = './/a[@class="ZLl54"]/@href';
+	const XPATH_EXPRESSION_ITEM_AUTHOR = './/span[@class="z0zUgf"][last()]';
+	const XPATH_EXPRESSION_ITEM_TIMESTAMP = './/div[@class="tRlaM"]';
+	const XPATH_EXPRESSION_ITEM_ENCLOSURES = '';
+	const XPATH_EXPRESSION_ITEM_CATEGORIES = '';
+	const SETTING_FIX_ENCODING = true;
+
+	protected function getSourceUrl() {
+		$source = self::URI;
+
+		$account = $this->getInput('account');
+		if($account) {
+			$source = $source . '/a/' . $account;
+		}
+		return $source . '/g/' . $this->getInput('group');
+	}
+
+	protected function provideWebsiteContent() {
+		return defaultLinkTo(getContents($this->getSourceUrl()), self::URI);
```

Comment:
The reason why this is necessary is because Google groups defines the base as the root (`<base href="https://groups.google.com/">`) while XPathAbstract uses the source url as the base (for example `https://groups.google.com/a/mozilla.org/g/announce`), which leads to duplication of `a/mozilla.org/g` in all the relative urls used in the document if this is not used to correctly convert the relative links first.